types/openebs.go

```diff
@@ -118,11 +128,12 @@ type Analytics struct {
 // component such as it it is enabled or not, no of
 // replicas, nodeselector, etc.
 type Component struct {
-	Enabled      *bool               `json:"enabled"`
-	Replicas     *int32              `json:"replicas"`
-	NodeSelector map[string]string   `json:"nodeSelector"`
-	Tolerations  []corev1.Toleration `json:"tolerations"`
-	Affinity     *corev1.Affinity    `json:"affinity"`
+	Enabled      *bool                        `json:"enabled"`
+	Replicas     *int32                       `json:"replicas"`
+	Resources    *corev1.ResourceRequirements `json:"resources"`
+	NodeSelector map[string]string            `json:"nodeSelector"`
```

Comment:
Are these already supported in old design or is this an enhancement?
I hope we are on same page to enable these orthogonal features via separate custom resources eventually.
Avoid changes to PUBLIC APIs else we need to support migration to newer design.